File: extrair_contrato.js
const fs = require("fs");
const path = require("path");
const AdmZip = require("adm-zip");

try {
  const docxPath = path.join(
    __dirname,
    "Contrato Fideliza_ESC_0016_ Modelo.docx",
  );

  // Verificar se arquivo existe
  if (!fs.existsSync(docxPath)) {
    console.log("❌ Arquivo não encontrado:", docxPath);
    process.exit(1);
  }

  console.log("📂 Abrindo arquivo:", docxPath);

  // Tentar usando adm-zip
  try {
    const zip = new AdmZip(docxPath);
    const xmlEntry = zip.getEntry("word/document.xml");

    if (!xmlEntry) {
      console.log("❌ Arquivo document.xml não encontrado no .docx");
      process.exit(1);
    }

    const xmlContent = zip.readAsText(xmlEntry);

    // Extrair texto usando regex simples
    const textMatches = xmlContent.match(/<w:t[^>]*>([^<]*)<\/w:t>/g) || [];
    const fullText = textMatches
      .map((m) => m.replace(/<w:t[^>]*>/, "").replace(/<\/w:t>/, ""))
      .join("");

    console.log("\n" + "=".repeat(100));
    console.log("CONTRATO EXTRAÍDO COM SUCESSO");
    console.log("=".repeat(100) + "\n");
    console.log(fullText);
    console.log("\n" + "=".repeat(100));

    // Salvar em arquivo
    fs.writeFileSync("contratos_conteudo_extraido.txt", fullText, "utf-8");
    console.log("\n✅ Salvo em: contratos_conteudo_extraido.txt");
  } catch (e) {
    console.log("❌ Erro ao processar com adm-zip:", e.message);
    process.exit(1);
  }
} catch (e) {
  console.log("❌ Erro:", e.message);
  process.exit(1);
}

File: extrair_contrato.py
import zipfile
import xml.etree.ElementTree as ET
import json

docx_path = r"Contrato Fideliza_ESC_0016_ Modelo.docx"

try:
    with zipfile.ZipFile(docx_path, 'r') as zip_ref:
        with zip_ref.open('word/document.xml') as xml_file:
            content = xml_file.read().decode('utf-8')
            
            # Parse XML
            root = ET.fromstring(content)
            
            # Namespace
            ns = {
                'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main',
                'wp': 'http://schemas.openxmlformats.org/drawingml/2006/wordprocessingDrawing',
                'a': 'http://schemas.openxmlformats.org/drawingml/2006/main',
                'pic': 'http://schemas.openxmlformats.org/drawingml/2006/picture'
            }
            
            # Extrair todo texto
            all_text = []
            for t in root.findall('.//w:t', ns):
                if t.text:
                    all_text.append(t.text)
            
            full_text = ''.join(all_text)
            
            # Salvar em arquivo legível
            with open('contratos_conteudo_extraido.txt', 'w', encoding='utf-8') as f:
                f.write(full_text)
            
            print("✅ Contrato extraído com sucesso!")
            print("\n" + "="*80)
            print(f"CONTEÚDO DO CONTRATO:\n")
            print("="*80)
            print(full_text)
            print("\n" + "="*80)
            
except Exception as e:
    print(f"❌ Erro: {e}")
    import traceback
    traceback.print_exc()
